pkg/cloudprovider/cloudprovider.go

```diff
@@ -58,5 +58,5 @@ func (k *k3s) ProviderName() string {
 }
 
 func (k *k3s) HasClusterID() bool {
-	return false
+	return true
```

Comment:
@galal-hussein is this reversing an intentional decision?

Reply:
As per kubernetes/cloud-provider#12 the ClusterID requirement was never really followed through on, so the flag is probably going to be removed in the future. Setting the flag and returning false here currently does nothing except cause a warning to be printed (I checked).

Reply:
yeah clusterID i believe  was only utilized by aws provider thats why we returned false, I am good with the change if it doesnt generate unwanted warnings/errors when starting the ccm